### ISSUE TITLE
way to see all the runs that targeted an asset, even if they didn't materialize it

### ISSUE THREAD

**Issue by sryza** [dagster-io/dagster#7963]
Reasons why a run might not materialize an asset that it targets:
* The run fails
* The code inside the run determines that the asset is up-to-date and doesn't need to be rematerialized

**Comment by bengotow** [comment]
I think we looked at doing this back when we started adding live data to the asset graph and there was a technical reason that it was challenging at the GraphQL layer. We landed on exposing just assetNode.latestRun, which leverages a `last_run_id` that is cached on to each `AssetEntry`. I think scanning the whole runs table for runs targeting a specific asset required a big refactor at the time? Maybe it'd be more feasible now!

**Comment by sryza** [comment]
I think it's likely that we added ASSET_MATERIALIZATION_PLANNED since then, which I believe should make this trivial. Thoughts @clairelin135 ?

**Comment by clairelin135** [comment]
Yep, I think this should be pretty trivial using materialization planned events. We can just query for materialization planned events for a given asset key and fetch the corresponding run IDs.
